### DISCUSSION TITLE
Debouncing form state updates for text inputs (both dirty and validation debounce)?

### DISCUSSION THREAD

**Discussion by briandonahue** [react-hook-form/react-hook-form#2455] [Q&A]
Hello,

I feel like this question has been asked a few different ways, but I am not sure I understand the answers or if they apply to my case. Apologies if I am repeating others' questions.  

I have a form that validates onChange. I am also running some logic when the form is dirty to a) figure out the next input to show the user,  and b) save back form answers to an API.

For text inputs, I would like to debounce BOTH isDirty and validation for a field, so that it is not triggered on every key press.  Is this possible?

Basically I would like the form state to not be updated until the debounce period. Any advice is greatly appreciated!

**Comment by bluebill1049** [comment]
I think you can `debounce` at the `validate` function within `register`.

**Comment by briandonahue** [comment]
Thank you for replying! I am seeing weird behavior with `debounce` where the validation message isn't showing up.  
```
  const validation = (value) => {
    if (validationRegex) {
      return (new RegExp(validationRegex).test(value)) || (validationMessage || "Invalid.")
    }
    return true
  }
  ```
Using this function directly works with `register` and `validate: validation`, the validation message shows.  But wrapping it in debounce like:
 ```
           register({
              validate: debounce(validation, 500)
              })
 ```
or 
 ```
           register({
              validate: value => debounce(validation, 500)(value)
              })
 ```
It seems to execute successfully (I can `console.log` the result and it logs out correctly) but the message does not display...

**Reply by bluebill1049** [reply]
can you reproduce this in a codesandbox?

**Reply by briandonahue** [reply]
Thank you - I will try to do that this week if I have time. Of course I have a deadline this week, so am going to have to just hold off on and hope I can keep my homegrown solution from falling over before the first production deployment 😄
